### PR TITLE
Added disallow hook to start to deprecate disable hook.

### PR DESCRIPTION
**Pull request by eddyystop**
https://github.com/feathersjs/feathers-hooks-common/issues/28 we agreed to create a new hook rather than change disable.

- disable hook has a predicate & we are deprecating predicates.
- the disable predicate is implemented in perhaps the wrong sense.
- In https://github.com/feathersjs/feathers-hooks-common/issues/28 we agreed to create a new hook rather than change disable.

### CODE REVIEW

File: src/services/disable-method.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const errors = feathersErrors.errors;
+
+export default function (...providers) {
+  return hook => {
+    const hookProvider = (hook.params || {}).provider;
+
+    const anyProvider = providers.length === 0;
+    const thisProvider = providers.some(provider =>
+      provider === hookProvider ||
+      (provider === 'server' && !hookProvider) ||
```
Comment (daffl) [inline src/services/disable-method.js]: The correct provider settings are:

- external: `!!provider`
- REST: `provider === 'rest'`
- Socket.io: `provider === 'socketio'`
- Primus: `provider === 'primus'`

File: README.md [hunk 2]
```diff
@@ -24,8 +24,7 @@ will be removed next version. Use the new `populate` hook.
 - Use `deleteByDot` rather than `setByDot(obj, path, value, true)`.
 - The `delete` hook should be used instead of `remove`.
 You will need to wrap `delete` it in an conditional if you want it to work like `remove` does.
-- The `stop` hook may be used instead of `disable`.
-The `disable` hook when `false` instead of the documented `true`.
+- The `disableMethod` hook may be used instead of `disable`.
```
Comment (daffl) [inline README.md]: Should we maybe call it `disallow` or `refuse`?

**Comment by eddyystop** [comment]
Both are much better names. I'd go with disallow to stay within the "dis"
family.

**Comment by eddyystop** [comment]
I prefer names that are clear about what they do, even if they get long. I would prefer a name like `disallowProvider`. What do you think?

**Comment by daffl** [comment]
I always think short names combined with the right parameters work. Something like `disallow('external')` or `disallow('rest')` is fairly explanatory to me. The `provider` postfix only makes sense if you know what a provider is (and I think @ekryski and @marshallswain wanted to change to calling it _Transport_).

**Comment by ekryski** [comment]
Since we are already having to do a major version and have breaking changes, we may as well just call it disable and perform the functionality we want to. Rip that bandaid off instead of having even more legacy code to maintain. That's my 2 cents anyway.

**Comment by ekryski** [comment]
Otherwise I agree with @daffl. Just go with `disallow`.

**Comment by eddyystop** [comment]
Waiting for a ship it then.

**Comment by ekryski** [comment]
:shipit:

**Comment by daffl** [comment]
:shipit:

**Pull request merged**